plugins/lsp/lsp-status.nix

```diff
@@ -0,0 +1,83 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "lsp-status";
+  originalName = "lsp-status.nvim";
+  url = "https://github.com/nvim-lua/lsp-status.nvim";
+  defaultPackage = pkgs.vimPlugins.lsp-status-nvim;
+  maintainers = [ helpers.maintainers.b3nb5n ];
+  settingsOptions = with lib; {
+    plugins.lsp-status = rec {
+      enable = mkEnableOption "lsp-status.nvim";
+
+      package = helpers.mkPluginPackageOption "lsp-status.nvim" pkgs.vimPlugins.lsp-status-nvim;
+
+      kind_labels = mkOption {
+        type = types.attrsOf types.str;
+        default = { };
+        description = "An optional map from LSP symbol kinds to label symbols. Used to decorate the current function name.";
+      };
+
+      select_symbol = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "An optional handler of the form function(cursor_pos, document_symbol) that should return true if document_symbol (a DocumentSymbol) should be accepted as the symbol currently containing the cursor.";
+      };
+
+      current_function = mkOption {
+        type = types.bool;
+        default = true;
+        description = "true if the current function should be updated and displayed in the default statusline component.";
+      };
+
+      show_filename = mkOption {
+        type = types.bool;
+        default = true;
+        description = "true if the current function should be updated and displayed in the default statusline component.";
+      };
+
+      indicator_ok = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "string to show as diagnostics. If you don't have Nerd/Awesome Fonts you can replace defaults with ASCII chars.";
+      };
+
+      indicator_errors = indicator_ok;
```

Comment:
Maybe declare common/shared descriptions in a `let` block instead of using `rec`